addons/website/tests/test_ui.py

```diff
@@ -124,6 +124,69 @@ def test_admin_tour_rte_translator(self):
         })
         self.start_tour(self.env['website'].get_client_action_url('/'), 'rte_translator', login='admin', timeout=120)
 
+    def test_translate_menu_name(self):
+        parseltongue = self.env['res.lang'].create({
+            'name': 'Parseltongue',
+            'code': 'pa_GB',
+            'iso_code': 'pa_GB',
+            'url_code': 'pa_GB',
+        })
+        self.env['res.lang']._activate_lang('pa_GB')
+        default_website = self.env.ref('website.default_website')
+        default_website.language_ids = self.env.ref('base.lang_en') + parseltongue
+        default_website.default_lang_id = self.env.ref('base.lang_en')
+        website_1_menu = self.env['website.menu'].search(['&', ('website_id', '=', 1), ('parent_id', '=', False)])
+        new_menu = self.env['website.menu'].create({
+            'name': 'Menu to edit',
+            'parent_id': website_1_menu.id,
+            'website_id': 1,
+            'url': '/englishURL',
+        })
+
+        # Timeout needed because it can take a long time to install a lang on a website.
+        self.start_tour(self.env['website'].get_client_action_url('/'), 'translate_menu_name', login='admin', timeout=120)
+
+        self.assertNotEqual(new_menu.name, 'value pa-GB', msg="The new menu should not have its value edited, only its translation")
+        translation = new_menu.get_field_translations('name', langs=['pa_GB'])
+        self.assertEqual(translation[0][0]['value'], 'value pa-GB', msg="The new translation should be set")
+
+    def test_snippet_translation(self):
+        fake_lang = self.env['res.lang'].create({
+            'name': 'Parseltongue',
+            'code': 'pa_GB',
+            'iso_code': 'pa_GB',
+            'url_code': 'pa_GB',
+        })
+        fake_user_lang = self.env['res.lang'].create({
+            'name': 'Fake User Lang',
+            'code': 'fu_GB',
+            'iso_code': 'fu_GB',
+            'url_code': 'fu_GB',
+        })
+        self.env["base.language.install"].create({
+            'overwrite': True,
+            'lang_ids': [(6, 0, [fake_lang.id, fake_user_lang.id])],
+        }).lang_install()
+        self.env.ref('base.user_admin').write({'lang': fake_user_lang.code})
+        self.env['ir.ui.view'].search([('key', '=', 'website.s_cover')]).update_field_translations('arch_db', {
```

Comment:
that's not the proper way to find ir.ui.view
Either you want the one from the xml file -> self.env.ref
Either you want to COW one -> `viewref` or other util we have to do that